diff --git a/includes/common.inc b/includes/common.inc
index 38ea92abe..d201e5bac 100644
--- a/includes/common.inc
+++ b/includes/common.inc
@@ -8072,6 +8072,11 @@ function entity_info_cache_clear() {
  *   - 2: Bundle name of the entity, or NULL if $entity_type has no bundles.
  */
 function entity_extract_ids($entity_type, $entity) {
+  // Make sure we have an entity.
+  if (empty($entity)) {
+    return [NULL, NULL, NULL];
+  }
+
   $info = entity_get_info($entity_type);

   // Objects being created might not have id/vid yet.
