diff --git a/html/includes/common.inc b/html/includes/common.inc
index 38ea92abe..d64b47229 100644
--- a/html/includes/common.inc
+++ b/html/includes/common.inc
@@ -8072,6 +8072,11 @@ function entity_info_cache_clear() {
  *   - 2: Bundle name of the entity, or NULL if $entity_type has no bundles.
  */
 function entity_extract_ids($entity_type, $entity) {
+  // Make sure we have an entity.
+  if (empty($entity)) {
+    return [NULL, NULL, NULL];
+  }
+
   $info = entity_get_info($entity_type);

   // Objects being created might not have id/vid yet.
@@ -8087,7 +8092,10 @@ function entity_extract_ids($entity_type, $entity) {
   if (!empty($info['entity keys']['bundle'])) {
     // Explicitly fail for malformed entities missing the bundle property.
     if (!isset($entity->{$info['entity keys']['bundle']}) || $entity->{$info['entity keys']['bundle']} === '') {
-      throw new EntityMalformedException(t('Missing bundle property on entity of type @entity_type.', array('@entity_type' => $entity_type)));
+      // Ignore this for taxonomy terms.
+      if ($entity_type != 'taxonomy_term') {
+        throw new EntityMalformedException(t('Missing bundle property on entity of type @entity_type.', array('@entity_type' => $entity_type)));
+      }
     }
     $bundle = $entity->{$info['entity keys']['bundle']};
   }
