diff --git a/redis/redis.path.inc b/redis/redis.path.inc
index 73ffa091..8d7531ac 100644
--- a/redis/redis.path.inc
+++ b/redis/redis.path.inc
@@ -235,6 +235,10 @@ function drupal_cache_system_paths() {
  *   found.
  */
 function drupal_get_path_alias($path = NULL, $path_language = NULL) {
+  // Patch to enable path_alias_xt.module to operate.
+  if (module_exists('path_alias_xt')) {
+    return path_alias_xt_get_path_alias($path, $path_language);
+  }
   // If no path is specified, use the current page's path.
   if ($path == NULL) {
     $path = $_GET['q'];
