diff --git a/date.module b/date.module
index 7415015a..9e9ccb9a 100644
--- a/date.module
+++ b/date.module
@@ -234,6 +234,9 @@ function date_formatter_process($formatter, $entity_type, $entity, array $field,
     $timezone = date_get_timezone($field['settings']['tz_handling'], $timezone);
   }
   $timezone_db = date_get_timezone_db($field['settings']['tz_handling']);
+  if ($field['settings']['tz_handling'] == 'date') {
+    $timezone_db = $timezone;
+  }
   $db_format = date_type_format($field['type']);
   $process = date_process_values($field);
   foreach ($process as $processed) {
