diff --git a/dbtng_migrator.batch.inc b/dbtng_migrator.batch.inc
index 52389b7..8defd65 100644
--- a/dbtng_migrator.batch.inc
+++ b/dbtng_migrator.batch.inc
@@ -108,11 +108,12 @@ function dbtng_migrator_batch_migrate_table($origin, $destination, &$context) {
   $schema = drupal_get_schema($table);

   // Set a default sort as range queries must have sorts.
-  $order_by = key($schema['fields']);
   if (isset($schema['primary key'])) {
-    $order_by = current($schema['primary key']);
+    $order_by = $schema['primary key'];
+  }
+  else {
+    $order_by = array_keys($schema['fields']);
   }
-

   try {
     // We have to revert to the default database to process a t() function call.
@@ -129,11 +130,15 @@ function dbtng_migrator_batch_migrate_table($origin, $destination, &$context) {

     while ($count < $total) {
       db_set_active($origin);
-      $rows = db_select($table, 't')
+      $query = db_select($table, 't')
                 ->fields('t', array_keys($schema['fields']))
-                ->orderBy($order_by)
-                ->range($count, $limit)
-                ->execute();
+                ->range($count, $limit);
+
+      foreach ($order_by as $order_by_field) {
+        $query->orderBy($order_by_field);
+      }
+
+      $rows = $query->execute()->fetchAll();

       // No need to generate insert queries that don't insert anything.
       if (!count($rows)) {
