diff --git a/modules/node/node.module b/modules/node/node.module
index 1d88834..8d42c42 100644
--- a/modules/node/node.module
+++ b/modules/node/node.module
@@ -961,6 +961,15 @@ function node_load_multiple($nids = array(), $conditions = array(), $reset = FAL
  *   A fully-populated node object, or FALSE if the node is not found.
  */
 function node_load($nid = NULL, $vid = NULL, $reset = FALSE) {
+  // Make sure nid is numeric.
+  if (!is_numeric($nid)) {
+    return FALSE;
+  }
+
+  // Ignore non-numeric vid.
+  if (!is_numeric($vid)) {
+    $vid = NULL;
+  }
   $nids = (isset($nid) ? array($nid) : array());
   $conditions = (isset($vid) ? array('vid' => $vid) : array());
   $node = node_load_multiple($nids, $conditions, $reset);
