diff --git a/plugins/facetapi/dependency_facet.inc b/plugins/facetapi/dependency_facet.inc
index b13e0d8..eb80c7e 100644
--- a/plugins/facetapi/dependency_facet.inc
+++ b/plugins/facetapi/dependency_facet.inc
@@ -30,7 +30,7 @@ class FacetapiDependencyFacet extends FacetapiDependency {
     else {
       // Show this facet if no facet dependencies are configured.
       $facets = array_filter($this->settings['facets']);
-      if (!count($facets)) {
+      if (empty($facets)) {
         return;
       }
       else {
@@ -45,7 +45,7 @@ class FacetapiDependencyFacet extends FacetapiDependency {
               $items_string = trim($this->settings['facet_items_' . $facet_name]);
               $facet_items = drupal_explode_tags($items_string);
               // Show/hide if dependency is met without specific dependency items.
-              if (empty($items_string) || !count($facet_items)) {
+              if (empty($items_string) || empty($facet_items)) {
                 return !$this->settings['reverse_dependency'];
               }
               // Show/hide if dependency is met with specific dependency items.
@@ -136,13 +136,13 @@ class FacetapiDependencyFacet extends FacetapiDependency {
    * but the enabled facets are not known yet (afaik, hence the add. property)
    */
   public function getDefaultSettings() {
-    if (!count($this->defaultSettings)) {
+    if (empty($this->defaultSettings)) {
       $this->defaultSettings = array(
-                                     'facets' => array(),
-                                     'reverse_dependency' => FALSE,
-                                     'force_deactivation' => TRUE,
-                                     'regex' => FALSE,
-                                    );
+        'facets' => array(),
+        'reverse_dependency' => FALSE,
+        'force_deactivation' => TRUE,
+        'regex' => FALSE,
+      );
     }
     return $this->defaultSettings;
   }
