diff --git a/plugins/display_renderers/panels_renderer_standard.class.php b/plugins/display_renderers/panels_renderer_standard.class.php
index 88ac1df..e037705 100644
--- a/plugins/display_renderers/panels_renderer_standard.class.php
+++ b/plugins/display_renderers/panels_renderer_standard.class.php
@@ -567,6 +567,9 @@ function render_pane(&$pane) {
     if (!empty($content->content)) {
       if (!empty($pane->style['style'])) {
         $style = panels_get_style($pane->style['style']);
+        if (!isset($pane->style['settings'])) {
+          $pane->style['settings'] = array();
+        }
 
         if (isset($style) && isset($style['render pane'])) {
           $output = theme($style['render pane'], array('content' => $content, 'pane' => $pane, 'display' => $this->display, 'style' => $style, 'settings' => $pane->style['settings']));
