diff --git a/html/sites/all/modules/contrib/shs/shs.module b/html/sites/all/modules/contrib/shs/shs.module
index 8f48742aa..56522fe71 100644
--- a/html/sites/all/modules/contrib/shs/shs.module
+++ b/html/sites/all/modules/contrib/shs/shs.module
@@ -489,8 +489,7 @@ function shs_field_widget_form(&$form, &$form_state, $field, $instance, $langcod
 
   // Prepare the list of options.
   if ($field['type'] == 'entityreference') {
-    // Get current selection handler.
-    $handler = entityreference_get_selection_handler($field, $instance, $element['#entity_type'], $element['#entity']);
+    // Add empty option.
     $options = array(
       '_none' => empty($element['#required']) ? t('- None -', array(), array('context' => 'shs')) : t('- Select a value -', array(), array('context' => 'shs')),
     );
@@ -547,6 +546,7 @@ function shs_field_widget_afterbuild($element, &$form_state) {
   if (!empty($default_value)) {
     // Use value from form_state (for example for fields with cardinality = -1).
     $element['#default_value'] = $default_value;
+    $element['#options'][$default_value] = $default_value;
   }
 
   // Add main Javascript behavior and style only once.
@@ -904,7 +904,19 @@ function shs_term_get_children($identifier, $parent = 0, array $settings = array
         // Only load entities if i18n_taxonomy or entity_translation is
         // installed.
         $load_entities = module_exists('i18n_taxonomy') || module_exists('entity_translation');
-        $tree = taxonomy_get_tree($vid, $parent, 1, $load_entities);
+
+        if ($load_entities) {
+          $tree_terms = taxonomy_get_tree($vid, $parent, 1);
+          $tree_tids = array();
+          foreach ($tree_terms as $tree_term) {
+            $tree_tids[] = $tree_term->tid;
+          }
+          $tree = taxonomy_term_load_multiple($tree_tids);
+        }
+        else {
+          $tree = taxonomy_get_tree($vid, $parent, 1);
+        }
+
         foreach ($tree as $term) {
           $term_name = $term->name;
           if (module_exists('i18n_taxonomy')) {
