diff --git a/modules/file/file.module b/modules/file/file.module
index 3d351fa..0d204c0 100644
--- a/modules/file/file.module
+++ b/modules/file/file.module
@@ -42,6 +42,8 @@ function file_menu() {
     'access arguments' => array('access content'),
     'theme callback' => 'ajax_base_page_theme',
     'type' => MENU_CALLBACK,
+    'file' => 'node.pages.inc',
+    'file path' => drupal_get_path('module', 'node'),
   );
   $items['file/progress'] = array(
     'page callback' => 'file_ajax_progress',
