diff --git a/search_api.admin.inc b/search_api.admin.inc
index a145ec7..077e94d 100644
--- a/search_api.admin.inc
+++ b/search_api.admin.inc
@@ -744,6 +744,12 @@ function search_api_admin_add_index(array $form, array &$form_state) {
         'This might have serious performance drawbacks and is generally not advised for larger sites.'),
     '#default_value' => FALSE,
   );
+  $form['options']['published_safe'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Published Safeguard'),
+    '#description' => t('Do not display published items in views using this index. Deactivate if you wish to do so, but remember to activate the Node Access filter.'),
+    '#default_value' => TRUE,
+  );
   $form['options']['cron_limit'] = array(
     '#type' => 'textfield',
     '#title' => t('Cron batch size'),
@@ -1249,6 +1255,12 @@ function search_api_admin_index_edit(array $form, array &$form_state, SearchApiI
       'invisible' => array(':input[name="read_only"]' => array('checked' => TRUE)),
     ),
   );
+  $form['options']['published_safe'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Published Safeguard'),
+    '#description' => t('Do not display published items in views using this index. Deactivate if you wish to do so, but remember to activate the Node Access filter.'),
+    '#default_value' => !empty($index->options['published_safe']),
+  );
   $form['options']['cron_limit'] = array(
     '#type' => 'textfield',
     '#title' => t('Cron batch size'),
diff --git a/search_api.module b/search_api.module
index 2661f34..a4d919d 100644
--- a/search_api.module
+++ b/search_api.module
@@ -2015,15 +2015,17 @@ function _search_api_query_add_node_access($account, SearchApiQueryInterface $qu
   }
 
   // Filter by the "published" status.
-  $published = $is_comment ? COMMENT_PUBLISHED : NODE_PUBLISHED;
-  if (!$is_comment && user_access('view own unpublished content')) {
-    $filter = $query->createFilter('OR');
-    $filter->condition('status', $published);
-    $filter->condition('author', $account->uid);
-    $query->filter($filter);
-  }
-  else {
-    $query->condition('status', $published);
+  if ($query->getIndex()->options['published_safe'] == TRUE) {
+    $published = $is_comment ? COMMENT_PUBLISHED : NODE_PUBLISHED;
+    if (!$is_comment && user_access('view own unpublished content')) {
+      $filter = $query->createFilter('OR');
+      $filter->condition('status', $published);
+      $filter->condition('author', $account->uid);
+      $query->filter($filter);
+    }
+    else {
+      $query->condition('status', $published);
+    }
   }
 
   // Filter by node access grants.
