diff --git a/includes/views/handlers/og_handler_filter_user_roles.inc b/includes/views/handlers/og_handler_filter_user_roles.inc
index c0573ee..4e07cb1 100644
--- a/includes/views/handlers/og_handler_filter_user_roles.inc
+++ b/includes/views/handlers/og_handler_filter_user_roles.inc
@@ -11,6 +11,7 @@ class og_handler_filter_user_roles extends views_handler_filter_many_to_one {
     $results = db_select('og_role', 'r')
       ->fields('r')
       ->groupBy('r.name')
+      ->groupBy('r.rid')
       ->execute();
     foreach ($results as $result) {
       $this->value_options[$result->name] = $result->name;
