diff --git a/og_menu.pages.inc b/og_menu.pages.inc
index 9728079..752646d 100644
--- a/og_menu.pages.inc
+++ b/og_menu.pages.inc
@@ -234,6 +234,15 @@ function og_menu_edit_item_form($form, &$form_state, $type, $group_type, $gid, $
      mpac_form_menu_edit_item_alter($form, $form_state);
   }
 
+  // Integrate with Menu Target if installed.
+  if (module_exists('menu_target')) {
+    module_load_include('inc', 'menu_target', 'menu_target.admin');
+    // Check function_exists(), just in case the include didn't load.
+    if (function_exists('menu_target_edit_form_extra_elements')) {
+      menu_target_edit_form_extra_elements($form, FALSE);
+    }
+  }
+
   return $form;
 }
 
