diff --git a/restful.module b/restful.module
index 54100b1..f0bfeb0 100644
--- a/restful.module
+++ b/restful.module
@@ -388,6 +388,10 @@ function restful_get_restful_handler_for_path($path = NULL) {
   list($major_version, $minor_version) = \RestfulBase::getVersionFromRequest($path);

   $handlers[$path] = restful_get_restful_handler($resource, $major_version, $minor_version);
+  // Make sure we have a fresh request.
+  $request = restful_parse_request();
+  $handlers[$path]->setRequest($request);
+
   return $handlers[$path];
 }

