diff --git a/js/shs.js b/js/shs.js
index 7d08ff6..cd62ff1 100644
--- a/js/shs.js
+++ b/js/shs.js
@@ -48,6 +48,7 @@
             var parent_id = 0;
             // Update class of wrapper element.
             $field.parent('.form-item').not('.shs-wrapper-processed').once('shs-wrapper');
+            $('<div class="ajax-progress" style="display:none;"><div class="throbber"></div></div>').appendTo($field.parent());
             // Create elements for each parent of the current value.
             $.each(fieldSettings.parents, function(index, parent) {
               level++;
@@ -118,6 +119,7 @@
       parent_value = [parent_value];
     }
 
+    $element.siblings('.ajax-progress').fadeIn();
     $.ajax({
       url: Drupal.settings.basePath + '?q=' + Drupal.settings.pathPrefix + 'js/shs/json',
       type: 'POST',
@@ -133,6 +135,7 @@
         }
       },
       success: function(data) {
+        $element.siblings('.ajax-progress').fadeOut();
         if (data.success == true) {
           if ($element.prop) {
             var options = $element.prop('options');
@@ -188,6 +191,7 @@
         }
       },
       error: function(xhr, status, error) {
+        $element.siblings('.ajax-progress').fadeOut();
       }
     });
   }
@@ -209,6 +213,7 @@
    *   Field settings.
    */
   termAddNew = function($triggering_element, $container, term, base_id, level, settings) {
+    $triggering_element.siblings('.ajax-progress').fadeIn();
     $.ajax({
       url: Drupal.settings.basePath + '?q=' + Drupal.settings.pathPrefix + 'js/shs/json',
       type: 'POST',
@@ -224,6 +229,7 @@
         }
       },
       success: function(data) {
+        $triggering_element.siblings('.ajax-progress').fadeOut();
         if (data.success == true && data.data.tid) {
           if ($triggering_element.prop) {
             var options = $triggering_element.prop('options');
@@ -264,6 +270,7 @@
         }
       },
       error: function(xhr, status, error) {
+        $triggering_element.siblings('.ajax-progress').fadeOut();
         // Reset value of triggering element.
         $triggering_element.val(0);
       },
