diff --git a/shs.module b/shs.module
index 8538e6a..036ae1d 100644
--- a/shs.module
+++ b/shs.module
@@ -364,8 +364,10 @@ function shs_field_widget_form(&$form, &$form_state, $field, $instance, $langcod
   // Get value.
   $element_value = NULL;
   $submitted_value = NULL;
-  if (!empty($form_state['values']) && !empty($element['#parents'])) {
-    $submitted_value = drupal_array_get_nested_value($form_state['values'], $element['#parents']);
+  if (!empty($form_state['values']) && !empty($element['#field_name']) && !empty($element['#language']) && isset($element['#delta'])) {
+    if ($field['type'] == 'entityreference' && isset($form_state['values'][$element['#field_name']][$element['#language']][$element['#delta']])) {
+      $submitted_value = $form_state['values'][$element['#field_name']][$element['#language']][$element['#delta']]['target_id'];
+    }
   }
   if (!empty($items[$delta][$field_column])) {
     // Use saved value from database or cache.
@@ -373,7 +375,7 @@ function shs_field_widget_form(&$form, &$form_state, $field, $instance, $langcod
   }
   elseif (!empty($submitted_value)) {
     // Use value from form_state (for example for fields with cardinality = -1).
-    $element_value = array('tid' => $submitted_value);
+    $element_value = $submitted_value;
   }
 
   // Get vocabulary names from allowed values.
@@ -415,14 +417,11 @@ function shs_field_widget_form(&$form, &$form_state, $field, $instance, $langcod
 
   // Prepare the list of options.
   if ($field['type'] == 'entityreference') {
-    // Get current selection handler.
-    $handler = entityreference_get_selection_handler($field, $instance, $element['#entity_type'], $element['#entity']);
-    $referencable_entities = $handler->getReferencableEntities();
     $options = array(
       '_none' => empty($element['#required']) ? t('- None -', array(), array('context' => 'shs')) : t('- Select a value -', array(), array('context' => 'shs')),
     );
-    foreach ($referencable_entities as $v_name => $terms) {
-      $options += $terms;
+    if (!empty($element_value)) {
+      $options[$element_value] = $element_value;
     }
   }
   else {
@@ -472,6 +471,7 @@ function shs_field_widget_afterbuild($element, &$form_state) {
   if (!empty($default_value)) {
     // Use value from form_state (for example for fields with cardinality = -1).
     $element['#default_value'] = $default_value;
+    $element['#options'][$default_value] = $default_value;
   }
 
   // Add main Javascript behavior and style only once.
